src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,133 @@
+package mate.academy.dao;
+
+import mate.academy.connection.util.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String AT_LEAST_ONE_ROW_EXCEPTION
+            = "Expected to insert at least one row, but inserted 0 rows";
+    private static final String ADD_BOOK_EXCEPTION = "Cannot add new book: ";
+    private static final String GET_BOOK_EXCEPTION = "Cannot get a book by id: ";
+    private static final String GET_ALL_BOOKS_EXCEPTION = "Cannot get books";
+    private static final String UPDATE_BOOK_EXCEPTION = "Cannot update book by id ";
+    private static final String DELETE_BOOK_EXCEPTION = "Cannot delete by id ";
+    private static final String NO_BOOK_EXCEPTION = "There are no book by id: ";
+    private static final String NULL_FIELDS_EXCEPTION = "Title name or price cannot be null";
+    private static final String NEGATIVE_ID_EXCEPTION = "ID cannot be negative";
+    private static final String NO_RECORDS_EXCEPTION = "There are no records in DB";
+    private static final int LESS_THAN_ONE = 1;
+    private static final int AT_LEAST_ONE = 0;
+    private static final int LESS_THAN_ZERO = 0;
+    private static final int ZERO = 0;
+
+    @Override
+    public Book create(Book book) {
+        if (book.getTitle() == null || book.getPrice() == null) {
```

Comment:
optional validation in this case
and as a rule, validation is not done at the dao level